src/azul/service/responseobjects/storage_service.py

```diff
@@ -1,34 +1,38 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
 from logging import getLogger
-from typing import Optional
+from typing import Optional, List
 import boto3
 from azul import config
 
 logger = getLogger(__name__)
+AWS_S3_DEFAULT_MINIMUM_PART_SIZE = 5242880  # 5 MB; see https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
+MULTIPART_UPLOAD_MAX_WORKERS = 12
```

Comment:
Didn't you conclude that 4 gave the fastest overall time?

Reply:
i forgot to commit that change.